File: hackhive-frontend/src/services/adminService.js
import api from "../api/axios";

async function request(promise) {
    const response = await promise;
    return response.data?.data ?? response.data;
}

export const adminService = {
    // Dashboard Stats
    getDashboardStatistics: () => request(api.get("/admin/dashboard")),

    // User Management
    getAllUsers: () => request(api.get("/admin/users")),
    getUserById: (userId) => request(api.get(`/admin/users/${userId}`)),
    getUsersByRole: (role) => request(api.get(`/admin/users/role/${role}`)),
    enableUser: (userId) => request(api.patch(`/admin/users/${userId}/enable`)),
    disableUser: (userId) => request(api.patch(`/admin/users/${userId}/disable`)),

    // Organizer Management
    getAllOrganizers: () => request(api.get("/admin/organizers")),
    getOrganizerById: (organizerProfileId) => request(api.get(`/admin/organizers/${organizerProfileId}`)),

    // Hackathon Management
    getAllHackathons: () => request(api.get("/admin/hackathons")),
    getHackathonById: (hackathonId) => request(api.get(`/admin/hackathons/${hackathonId}`)),
    deleteHackathon: (hackathonId) => request(api.delete(`/admin/hackathons/${hackathonId}`)),

    // Team Oversight
    getAllTeams: () => request(api.get("/admin/teams")),
    getTeamById: (teamId) => request(api.get(`/admin/teams/${teamId}`)),

    // Registration Oversight
    getAllRegistrations: () => request(api.get("/admin/registrations")),
    getRegistrationById: (registrationId) => request(api.get(`/admin/registrations/${registrationId}`)),
};

File: hackhive-backend/src/main/java/com/hackhive/admin/service/AdminService.java
package com.hackhive.admin.service;

import com.hackhive.admin.dto.response.AdminDashboardResponse;
import com.hackhive.admin.dto.response.AdminHackathonResponse;
import com.hackhive.admin.dto.response.AdminOrganizerResponse;
import com.hackhive.admin.dto.response.AdminRegistrationResponse;
import com.hackhive.admin.dto.response.AdminTeamResponse;
import com.hackhive.admin.dto.response.AdminUserResponse;

import java.util.List;

public interface AdminService {

    // Dashboard
    AdminDashboardResponse getDashboardStatistics();

    // User Management
    List<AdminUserResponse> getAllUsers();

    AdminUserResponse getUserById(
            Long userId
    );

    List<AdminUserResponse> getUsersByRole(
            String role
    );

    AdminUserResponse enableUser(
            Long userId
    );

    AdminUserResponse disableUser(
            Long userId
    );

    // Organizer Management
    List<AdminOrganizerResponse> getAllOrganizers();

    AdminOrganizerResponse getOrganizerById(
            Long organizerProfileId
    );

    AdminOrganizerResponse verifyOrganizer(
            Long organizerProfileId
    );

    AdminOrganizerResponse unverifyOrganizer(
            Long organizerProfileId
    );

    // Hackathon Management
    List<AdminHackathonResponse> getAllHackathons();

    AdminHackathonResponse getHackathonById(
            Long hackathonId
    );

    void deleteHackathon(
            Long hackathonId
    );

    // Team Oversight
    List<AdminTeamResponse> getAllTeams();

    AdminTeamResponse getTeamById(
            Long teamId
    );

    // Registration Oversight
    List<AdminRegistrationResponse> getAllRegistrations();

    AdminRegistrationResponse getRegistrationById(
            Long registrationId
    );
}